### ISSUE TITLE
Add support to store metadata

### ISSUE THREAD

**Issue by ghost** [palkan/logidze#60]
Logidze is really awesome but it can't replace papertrail in a project I'm working on.
This kind of stuff does not belong to the resource itself, it would be more pleasant to store it alongside the log.

```ruby
# application_controller.rb
def info_for_paper_trail
  { ip: request.remote_ip }
end

# model.rb
attr_accessor :comment
[...]
has_paper_trail meta: { reason: comment }
```

**Comment by anand180** [comment]
I'm running into the same limitation on my end. My work-around at the moment is to use `responsible_id` as it can take an arbitrary string as its value

```ruby
# store meta
Logidze.with_responsible({test: 123}.to_json) do
  Model.update(...)
end

# retrieve meta
meta = JSON.parse(log_data.responsible_id)
```

However, a better API would be something like
```ruby
# store meta
Logidze.with_meta({test: 123}) do
  Model.update(...)
end

# retrieve meta
meta = log_data.meta # => {test: 123}
```

**Comment by anand180** [comment]
If there is some consensus on what the API should look like, I don't mind helping out with the implementation.

**Comment by palkan** [comment]
I like the proposed API (`with_meta`); we can implement it the same way as `with_responsible`.

```ruby
record = Model.find(123)

Logidze.with_meta(test: 123) do
  record.update(attrs)
end

record.reload

# log_meta is equal to log_data.meta
meta = record.log_meta # => { "test" => 123 }
```

Note that the resulted Hash would have stringified keys (due to JSON encoding/decoding).

Feel free to propose a PR with the implementation!

**Comment by palkan** [comment]
Closed by #79 and released as 0.7.0
